test/test_unary_ufuncs.py

```diff
@@ -686,34 +686,25 @@ def test_abs_angle_complex_to_float(self, device, dtype):
                     torch.float32 if dtype is torch.complex64 else torch.float64
                 )
                 np_float_out = np_fn(a).astype(torch_to_numpy_dtype_dict[float_dtype])
-                float_out = torch.empty_like(t).float()
+                float_out = torch.empty_like(t, dtype=float_dtype)
                 torch_fn(t, out=float_out)
-                # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-                self.assertEqualIgnoreType(
-                    torch.from_numpy(np_float_out), float_out.cpu()
-                )
+                self.assertEqual(torch.from_numpy(np_float_out), float_out.cpu())
 
                 # Tests float out (resized out)
                 float_out = torch.empty(1, device=device, dtype=float_dtype)
                 torch_fn(t, out=float_out)
                 self.assertEqual(torch.from_numpy(np_float_out), float_out.cpu())
 
                 # Tests complex out
-                np_complex_out = np_fn(a)
+                np_complex_out = np_fn(a).astype(torch_to_numpy_dtype_dict[dtype])
```

Comment:
For the subsequent two "complex out" tests, I'm assuming that the expected behavior of `torch_fn(t, out=complex_out)` is to populate the `complex_out` tensor with the output, _keeping the dtype of `complex_out`_. 

Accordingly, for proper type checking, we must cast  `np_complex_out` to the `dtype` of `complex_out` (which happens to be the variable `dtype` in this scope).